addons/mrp/migrations/10.0.2.0/pre-migration.py

```diff
@@ -50,17 +50,29 @@ def rename_mrp_workorder(cr):
     renamed model.
     """
     tools.drop_view_if_exists(cr, 'mrp_workorder')
-    openupgrade.rename_tables(
-        cr, [('mrp_production_workcenter_line', 'mrp_workorder')]
-    )
     openupgrade.logged_query(
         cr,
         "DELETE FROM ir_model WHERE model = %s",
-        ('mrp.workorder', )
+        ('mrp.workorder',)
+    )
+    openupgrade.rename_tables(
+        cr, [('mrp_production_workcenter_line', 'mrp_workorder')]
     )
     openupgrade.rename_models(
         cr, [('mrp.production.workcenter.line', 'mrp.workorder')]
```

Comment:
For some reason I'm now failing migration with:
```
OpenUpgrade: mrp: pre-migration script called with version 9.0.1.1
OpenUpgrade: Running DELETE FROM ir_model WHERE model = mrp.workorder
OpenUpgrade: 1 rows affected
OpenUpgrade: table mrp_production_workcenter_line: renaming to mrp_workorder
OpenUpgrade: table mrp_production_workcenter_line_id_seq: renaming to mrp_workorder_id_seq
OpenUpgrade: model mrp.production.workcenter.line: renaming to mrp.workorder
odoo.sql_db: bad query: UPDATE ir_model_data SET name='model_mrp_workorder' WHERE name='model_mrp_production_workcenter_line' AND model = 'ir.model'
mrp: error in migration script mrp/migrations/10.0.2.0/pre-migration.py: IntegrityError('duplicate key value violates unique constraint "ir_model_data_module_name_uniq"\nDETAIL:  Key (name, module)=(model_mrp_workorder, mrp) already exists.\n',)
OpenUpgrade: duplicate key value violates unique constraint "ir_model_data_module_name_uniq"
DETAIL:  Key (name, module)=(model_mrp_workorder, mrp) already exists.
Traceback (most recent call last):
  File "/var/lib/odoo/openupgrade/openupgradelib/openupgradelib/openupgrade.py", line 1427, in wrapped_function
    if use_env2 else cr, version)
  File "/var/lib/odoo/openupgrade/10.0/addons/mrp/migrations/10.0.2.0/pre-migration.py", line 90, in migrate
    rename_mrp_workorder(cr)
  File "/var/lib/odoo/openupgrade/10.0/addons/mrp/migrations/10.0.2.0/pre-migration.py", line 62, in rename_mrp_workorder
    cr, [('mrp.production.workcenter.line', 'mrp.workorder')]
  File "/var/lib/odoo/openupgrade/openupgradelib/openupgradelib/openupgrade.py", line 548, in rename_models
    ('model_' + _new, 'model_' + _old),
  File "/var/lib/odoo/openupgrade/10.0/server/odoo/sql_db.py", line 154, in wrapper
    return f(self, *args, **kwargs)
  File "/var/lib/odoo/openupgrade/10.0/server/odoo/sql_db.py", line 231, in execute
    res = self._obj.execute(query, params)
IntegrityError: duplicate key value violates unique constraint "ir_model_data_module_name_uniq"
DETAIL:  Key (name, module)=(model_mrp_workorder, mrp) already exists.
```

I don't recall it failing back when I tested these fixed at end of January.. Odd.

Anyhow, it seems I still have a stale entry about mrp.workorder view in ir_model_data table as well (I have no cascade delete rules on this from ir_model in my DB at least), so need to delete that as well, in addition to ir_model above. Something like
```
openupgrade.logged_query(
    cr,
    "DELETE FROM ir_model_data WHERE name = %s",
    ('model_mrp_workorder',)
)
```

I'm not sure what changed meanwhile, maybe some openupgrade change (I see some small improvements in that area, but didn't look deep if they could affect this).
It does feel like there should be some openupgrade helper method to call for this use case (dropping a view that had an Odoo model backing it, or perhaps just together with a full model delete that checks if it's a view or table), instead of handrolling it here. Maybe there is something suitable for the ir_model + ir_model_data munging already too?

Reply:
After applying that quick ir_model_data "fix", the next failure related to this appears (this was definitely changed meanwhile in openupgradelib, but maybe just the way the query is done to be without regexp maybe?):

```
odoo.modules.migration: module mrp: Running migration [>10.0.2.0] pre-migration
OpenUpgrade: mrp: pre-migration script called with version 9.0.1.1
OpenUpgrade: Running DELETE FROM ir_model WHERE model = mrp.workorder
OpenUpgrade: 1 rows affected
OpenUpgrade: Running DELETE FROM ir_model_data WHERE name = model_mrp_workorder
OpenUpgrade: 1 rows affected
OpenUpgrade: table mrp_production_workcenter_line: renaming to mrp_workorder
OpenUpgrade: table mrp_production_workcenter_line_id_seq: renaming to mrp_workorder_id_seq
OpenUpgrade: model mrp.production.workcenter.line: renaming to mrp.workorder
odoo.sql_db: bad query: UPDATE ir_model_data imd
            SET name = 'field_' || 'mrp_workorder' || '_' || imf.name
            FROM ir_model_fields imf
            WHERE imd.model = 'ir.model.fields'
                AND imd.name = 'field_' || 'mrp_production_workcenter_line' || '_' || imf.name
                AND imf.model = 'mrp.production.workcenter.line'
OpenUpgrade: mrp: error in migration script mrp/migrations/10.0.2.0/pre-migration.py: IntegrityError('duplicate key value violates unique constraint "ir_model_data_module_name_uniq"\nDETAIL:  Key (name, module)=(field_mrp_workorder_id, mrp) already exists.\n',)
OpenUpgrade: duplicate key value violates unique constraint "ir_model_data_module_name_uniq"
DETAIL:  Key (name, module)=(field_mrp_workorder_id, mrp) already exists.
Traceback (most recent call last):
  File "/var/lib/odoo/openupgrade/openupgradelib/openupgradelib/openupgrade.py", line 1427, in wrapped_function
    if use_env2 else cr, version)
  File "/var/lib/odoo/openupgrade/10.0/addons/mrp/migrations/10.0.2.0/pre-migration.py", line 95, in migrate
    rename_mrp_workorder(cr)
  File "/var/lib/odoo/openupgrade/10.0/addons/mrp/migrations/10.0.2.0/pre-migration.py", line 67, in rename_mrp_workorder
    cr, [('mrp.production.workcenter.line', 'mrp.workorder')]
  File "/var/lib/odoo/openupgrade/openupgradelib/openupgradelib/openupgrade.py", line 557, in rename_models
    (AsIs(_new), AsIs(_old), old),
  File "/var/lib/odoo/openupgrade/10.0/server/odoo/sql_db.py", line 154, in wrapper
    return f(self, *args, **kwargs)
  File "/var/lib/odoo/openupgrade/10.0/server/odoo/sql_db.py", line 231, in execute
    res = self._obj.execute(query, params)
IntegrityError: duplicate key value violates unique constraint "ir_model_data_module_name_uniq"
DETAIL:  Key (name, module)=(field_mrp_workorder_id, mrp) already exists.
```

Reply:
Solved that one with
```
openupgrade.logged_query(
    cr,
    "DELETE FROM ir_model_data WHERE model = %s AND module = %s and name LIKE %s",
    ('ir.model.fields', 'mrp', 'field_mrp_workorder_%',)
)
```

but clearly it could use a helper. Additionally there's old properties that might remain stale and potentially clash with new models with the same name, which could use proper removing for model deletion. As this wasn't a case with old mrp.workorder view, I didn't look into doing that here; got past this now and completed a migration test finally again with these two additional DELETE FROM commands.